pool/workers.go

```diff
@@ -0,0 +1,139 @@
+package pool
+
+import (
+	"time"
+
+	"github.com/serverless/event-gateway-connector/connection"
+	"go.uber.org/zap"
+)
+
+type workerMap map[int]*worker
+
+// worker is the internal representation of the worker process
+type worker struct {
+	id     int
+	recv   chan *connection.Connection
+	errors chan workerError
+	close  chan int
+	conn   *connection.Connection
+	log    *zap.SugaredLogger
+	done   chan bool
+}
+
+// workerError for cases where the worker ends up failing for a specific reason
+type workerError struct {
+	id  int
+	err error
+}
+
+// StartWorkers receives the number of worker goroutines from the main process
+func StartWorkers(numWorkers int, conns chan *connection.Connection, done <-chan bool) error {
```

Comment:
Also, you need to accept chan with `watcher.Event` here because in case of deleting subscription you need to stop workers. We don't need to support deletion right now but that should be the approach. I will change `watcher.Event` to provide Connection object no `[]byte`.